### ISSUE TITLE
`pixi list` and `pixi tree` behave weird when specifying the same dependency in conda and pypi

### ISSUE THREAD

**Issue by pavelzw** [prefix-dev/pixi#1158]
```toml
# pyproject.toml
[project]
name = "tmp.BQA3Arxqvx"
requires-python = ">=3.9"
version = "0.1.0"
dependencies = [
    "polars>=0.14"
]

[tool.pixi.project]
channels = ["conda-forge"]
platforms = ["linux-64", "osx-arm64"]

[tool.pixi.dependencies]
polars = ">=0.14"
```

```
❯ pixi list
Package          Version       Build                 Size       Kind   Source
bzip2            1.0.8         h93a5062_5            119.5 KiB  conda  bzip2-1.0.8-h93a5062_5.conda
ca-certificates  2024.2.2      hf0a4a13_0            152.1 KiB  conda  ca-certificates-2024.2.2-hf0a4a13_0.conda
libblas          3.9.0         22_osxarm64_openblas  14.5 KiB   conda  libblas-3.9.0-22_osxarm64_openblas.conda
libcblas         3.9.0         22_osxarm64_openblas  14.4 KiB   conda  libcblas-3.9.0-22_osxarm64_openblas.conda
libcxx           16.0.6        h4653b0c_0            1.1 MiB    conda  libcxx-16.0.6-h4653b0c_0.conda
libexpat         2.6.2         hebf3989_0            62.2 KiB   conda  libexpat-2.6.2-hebf3989_0.conda
libffi           3.4.2         h3422bc3_5            38.1 KiB   conda  libffi-3.4.2-h3422bc3_5.tar.bz2
libgfortran      5.0.0         13_2_0_hd922786_3     107.6 KiB  conda  libgfortran-5.0.0-13_2_0_hd922786_3.conda
libgfortran5     13.2.0        hf226fd6_3            974 KiB    conda  libgfortran5-13.2.0-hf226fd6_3.conda
liblapack        3.9.0         22_osxarm64_openblas  14.4 KiB   conda  liblapack-3.9.0-22_osxarm64_openblas.conda
libopenblas      0.3.27        openmp_h6c19121_0     2.8 MiB    conda  libopenblas-0.3.27-openmp_h6c19121_0.conda
libsqlite        3.45.2        h091b4b1_0            806 KiB    conda  libsqlite-3.45.2-h091b4b1_0.conda
libzlib          1.2.13        h53f4e23_5            47 KiB     conda  libzlib-1.2.13-h53f4e23_5.conda
llvm-openmp      18.1.3        hcd81f8e_0            269.8 KiB  conda  llvm-openmp-18.1.3-hcd81f8e_0.conda
ncurses          6.4.20240210  h078ce10_0            801 KiB    conda  ncurses-6.4.20240210-h078ce10_0.conda
numpy            1.26.4        py312h8442bc7_0       5.8 MiB    conda  numpy-1.26.4-py312h8442bc7_0.conda
openssl          3.2.1         h0d3ecfb_1            2.7 MiB    conda  openssl-3.2.1-h0d3ecfb_1.conda
packaging        24.0          pyhd8ed1ab_0          48.7 KiB   conda  packaging-24.0-pyhd8ed1ab_0.conda
polars           0.20.19       py312he03d2e2_0       16.1 MiB   conda  polars-0.20.19-py312he03d2e2_0.conda
polars           0.20.19                                        pypi   https://files.pythonhosted.org/packages/ab/8e/01084c7c42885e05729d34d696dd17c30bf51467588265ed843fc37c7be9/polars-0.20.19-cp38-abi3-macosx_11_0_arm64.whl
python           3.12.2        hdf0ec26_0_cpython    12.5 MiB   conda  python-3.12.2-hdf0ec26_0_cpython.conda
python_abi       3.12          4_cp312               6.4 KiB    conda  python_abi-3.12-4_cp312.conda
readline         8.2           h92ec313_1            244.5 KiB  conda  readline-8.2-h92ec313_1.conda
tk               8.6.13        h5083fa2_1            3 MiB      conda  tk-8.6.13-h5083fa2_1.conda
tzdata           2024a         h0c530f3_0            117 KiB    conda  tzdata-2024a-h0c530f3_0.conda
xz               5.2.6         h57fd34a_0            230.2 KiB  conda  xz-5.2.6-h57fd34a_0.tar.bz2
```

```
❯ pixi tree
├── polars 0.20.19
├── python 3.12.2
│   ├── bzip2 1.0.8
│   ├── libexpat 2.6.2
│   ├── libffi 3.4.2
│   ├── libsqlite 3.45.2
│   │   └── libzlib 1.2.13
│   ├── libzlib 1.2.13 (*)
│   ├── ncurses 6.4.20240210
│   ├── openssl 3.2.1
│   │   └── ca-certificates 2024.2.2
│   ├── readline 8.2
│   │   └── ncurses 6.4.20240210 (*)
│   ├── tk 8.6.13
│   │   └── libzlib 1.2.13 (*)
│   ├── tzdata 2024a
│   └── xz 5.2.6
└── polars 0.20.19
```

Note how polars is listed once as a pypi and once as a conda dependency. I would have expected only the conda dependency.
Also note how in `pixi tree` polars occurs twice.
And that the `polars` entry in `pixi tree` is incomplete: it's missing `numpy` for example.

Compare to a `pixi tree` when using only conda dependencies (i.e. remove `dependencies = ["polars>=0.14"]`)
```
❯ pixi tree
├── polars 0.20.19
│   ├── numpy 1.26.4
│   │   ├── libblas 3.9.0
│   │   │   └── libopenblas 0.3.27
│   │   │       ├── libgfortran 5.0.0
│   │   │       │   └── libgfortran5 13.2.0
│   │   │       │       └── llvm-openmp 18.1.3
│   │   │       ├── libgfortran5 13.2.0 (*)
│   │   │       └── llvm-openmp 18.1.3 (*)
│   │   ├── libcblas 3.9.0
│   │   │   └── libblas 3.9.0 (*)
│   │   ├── libcxx 16.0.6
│   │   ├── liblapack 3.9.0
│   │   │   └── libblas 3.9.0 (*)
│   │   ├── python 3.12.2
│   │   │   ├── bzip2 1.0.8
│   │   │   ├── libexpat 2.6.2
│   │   │   ├── libffi 3.4.2
│   │   │   ├── libsqlite 3.45.2
│   │   │   │   └── libzlib 1.2.13
│   │   │   ├── libzlib 1.2.13 (*)
│   │   │   ├── ncurses 6.4.20240210
│   │   │   ├── openssl 3.2.1
│   │   │   │   └── ca-certificates 2024.2.2
│   │   │   ├── readline 8.2
│   │   │   │   └── ncurses 6.4.20240210 (*)
│   │   │   ├── tk 8.6.13
│   │   │   │   └── libzlib 1.2.13 (*)
│   │   │   ├── tzdata 2024a
│   │   │   └── xz 5.2.6
│   │   └── python_abi 3.12
│   ├── packaging 24.0
│   │   └── python 3.12.2 (*)
│   ├── python 3.12.2 (*)
│   └── python_abi 3.12 (*)
└── python 3.12.2
    ├── bzip2 1.0.8 (*)
    ├── libexpat 2.6.2 (*)
    ├── libffi 3.4.2 (*)
    ├── libsqlite 3.45.2 (*)
    ├── libzlib 1.2.13 (*)
    ├── ncurses 6.4.20240210 (*)
    ├── openssl 3.2.1 (*)
    ├── readline 8.2 (*)
    ├── tk 8.6.13 (*)
    ├── tzdata 2024a (*)
    └── xz 5.2.6 (*)
```

**Comment by ruben-arts** [comment]
The `tree` indeed works  acts a bit strange for me to:
```
➜ pixi tree
├── polars 0.20.19 
│   ├── libgcc-ng 13.2.0 
│   │   ├── _libgcc_mutex 0.1 
│   │   └── _openmp_mutex 4.5 
...
│   └── python_abi 3.12 (*)
├── python 3.12.2 
│   ├── bzip2 1.0.8 (*)
│   ├── ld_impl_linux-64 2.40 (*)
│   ├── libexpat 2.6.2 (*)...
│   └── xz 5.2.6 (*)
└── polars 0.20.19 
    ├── libgcc-ng 13.2.0 (*)
    ├── numpy 1.26.4 (*)
    ├── packaging 24.0 (*)
    ├── python 3.12.2 (*)
    └── python_abi 3.12 (*)
```
As it shows polars twice.

But the list issue is part of the wrong mapping. You are actually installing both packages currently and that is wrong. 
We're fixing that ASAP!

**Comment by abkfenris** [comment]
Right now `tree` is getting the direct dependencies from what's recorded in the manifest for both Conda and PyPI packages. Is there a way to get which ones UV actually solves for? Or do I need to dig into all the new mapping work?

**Comment by ruben-arts** [comment]
We can possibly already use more of the information from the lockfile. As you know which kind of dependencies belongs to which name. If the direct dependencies would just contain a complete package you could print or filter them in a different way.

**Comment by abkfenris** [comment]
I think I've got a fix for `tree`, but I'm not seeing dupes with `list` on latest.

```sh
cargo run list --manifest-path ../pixi-test/1158/pyproject.toml
    Finished dev [unoptimized + debuginfo] target(s) in 0.49s
     Running `target/debug/pixi list --manifest-path ../pixi-test/1158/pyproject.toml`
Package          Version       Build                 Size       Kind   Source
bzip2            1.0.8         h93a5062_5            119.5 KiB  conda  bzip2-1.0.8-h93a5062_5.conda
ca-certificates  2024.2.2      hf0a4a13_0            152.1 KiB  conda  ca-certificates-2024.2.2-hf0a4a13_0.conda
libblas          3.9.0         22_osxarm64_openblas  14.5 KiB   conda  libblas-3.9.0-22_osxarm64_openblas.conda
libcblas         3.9.0         22_osxarm64_openblas  14.4 KiB   conda  libcblas-3.9.0-22_osxarm64_openblas.conda
libcxx           16.0.6        h4653b0c_0            1.1 MiB    conda  libcxx-16.0.6-h4653b0c_0.conda
libexpat         2.6.2         hebf3989_0            62.2 KiB   conda  libexpat-2.6.2-hebf3989_0.conda
libffi           3.4.2         h3422bc3_5            38.1 KiB   conda  libffi-3.4.2-h3422bc3_5.tar.bz2
libgfortran      5.0.0         13_2_0_hd922786_3     107.6 KiB  conda  libgfortran-5.0.0-13_2_0_hd922786_3.conda
libgfortran5     13.2.0        hf226fd6_3            974 KiB    conda  libgfortran5-13.2.0-hf226fd6_3.conda
liblapack        3.9.0         22_osxarm64_openblas  14.4 KiB   conda  liblapack-3.9.0-22_osxarm64_openblas.conda
libopenblas      0.3.27        openmp_h6c19121_0     2.8 MiB    conda  libopenblas-0.3.27-openmp_h6c19121_0.conda
libsqlite        3.45.2        h091b4b1_0            806 KiB    conda  libsqlite-3.45.2-h091b4b1_0.conda
libzlib          1.2.13        h53f4e23_5            47 KiB     conda  libzlib-1.2.13-h53f4e23_5.conda
llvm-openmp      18.1.3        hcd81f8e_0            269.8 KiB  conda  llvm-openmp-18.1.3-hcd81f8e_0.conda
ncurses          6.4.20240210  h078ce10_0            801 KiB    conda  ncurses-6.4.20240210-h078ce10_0.conda
numpy            1.26.4        py312h8442bc7_0       5.8 MiB    conda  numpy-1.26.4-py312h8442bc7_0.conda
openssl          3.2.1         h0d3ecfb_1            2.7 MiB    conda  openssl-3.2.1-h0d3ecfb_1.conda
packaging        24.0          pyhd8ed1ab_0          48.7 KiB   conda  packaging-24.0-pyhd8ed1ab_0.conda
polars           0.20.19       py312he03d2e2_0       16.1 MiB   conda  polars-0.20.19-py312he03d2e2_0.conda
python           3.12.2        hdf0ec26_0_cpython    12.5 MiB   conda  python-3.12.2-hdf0ec26_0_cpython.conda
python_abi       3.12          4_cp312               6.4 KiB    conda  python_abi-3.12-4_cp312.conda
readline         8.2           h92ec313_1            244.5 KiB  conda  readline-8.2-h92ec313_1.conda
tk               8.6.13        h5083fa2_1            3 MiB      conda  tk-8.6.13-h5083fa2_1.conda
tzdata           2024a         h0c530f3_0            117 KiB    conda  tzdata-2024a-h0c530f3_0.conda
xz               5.2.6         h57fd34a_0            230.2 KiB  conda  xz-5.2.6-h57fd34a_0.tar.bz2
```

**Comment by abkfenris** [comment]
Ok, PR up that fixes `tree` for at least more simple cases: #1184
